lib/build-components/build_components/configuration.py

```diff
@@ -0,0 +1,134 @@
+import dataclasses
+import functools
+import pathlib
+import typing
+
+import yaml
+
+
+class Undefined:
```

Comment:
This might be a good candidate to be a `@singleton`

Reply:
How about if I renamed it to `_Undefined` to at least hint that it shouldn't be instantiated outside of this module and add a comment to that effect? I can think of other means of creating a throwaway class to generate a sentinel object like this, but it seems a bit overkill?

Reply:
`del _Undefined` to the rescue